rails_application/app/controllers/supplies_controller.rb

```diff
@@ -1,10 +1,31 @@
 class SuppliesController < ApplicationController
+  class SupplyForm
+    include ActiveModel::Model
+    include ActiveModel::Validations
+
+    attr_reader :product_id, :quantity
+
+    def initialize(params)
+      @product_id = params[:product_id]
+      @quantity = params[:quantity]
+    end
```

Comment:
I didn't use `ActiveModel::Attributes` like in other places because then `only_integer: true` does not work as expected. It seems like `ActiveModel::Attributes` converts the value to integer during the initialization of the object. This is problematic because `"abc".to_i # => 0`, so the validation error was "Quantity must be greater than 0" instead of "Quantity is not a number".